### ISSUE TITLE
dbt templater not compatible with dbt-core version 1.4+

### ISSUE THREAD

**Issue by wtfzambo** [sqlfluff/sqlfluff#4457]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

dbt-core version 1.4 renamed all their exceptions in [this commit](https://github.com/dbt-labs/dbt-core/commit/eb200b46877c31469f410b7d47dfd245473ed7cb).

Since some of them are used inside `sqlfluff_templater_dbt`, trying to run `sqlfluff` with dbt plugin installed ends in an error, e.g.:

<details>
<summary>Screenshot</summary>

![image](https://user-images.githubusercontent.com/47462792/222793787-29dbbd48-f8d9-4be0-aa14-1700e0072463.png)

</details>

### Expected Behaviour

sqlfluff runs normally when using  `sqlfluff_templater_dbt` plugin in a dbt project

### Observed Behaviour

sqlfluff crashes when using  `sqlfluff_templater_dbt` plugin in a dbt project

### How to reproduce

In an empty folder:

```bash
python -m venv .venv
source .venv/bin/activate

pip install dbt-bigquery sqlfluff sqlfluff-templater-dbt

dbt init
# follow the instructions
```

Then in the root of the dbt project:
```bash
sqlfluff lint
```

### Dialect

bigquery

### Version

sqlfluff = 1.4.5
sqlfluff-templater-dbt = 1.4.5
python = 3.10.4

### Configuration

.sqlfluff
```ini
[sqlfluff]
templater = dbt
dialect = bigquery

[sqlfluff:templater:dbt]
project_dir = ./

[sqlfluff:layout:type:comma]
line_position = leading

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L030]
capitalisation_policy = upper
```

.sqlfluffignore
```
target/
dbt_packages/
macros/
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by pdebelak** [comment]
This is a duplicate issue and has been fixed in #4317 . You can install the latest pre-release version 2.0.0a5 to get this fix.

**Comment by wtfzambo** [comment]
Neat, thanks! I couldn't find it in the issues, I guess my search-fu has become rusty.
